### PR TITLE
Disallow Basic Auth over HTTP on Unix

### LINKED ISSUE #6779: Explicitly disallow Basic Auth over HTTP with PSRP on Linux
A security review of the proposed fix for https://github.com/PowerShell/psl-omi-provider/issues/114 (Basic Auth over HTTP does not work from PowerShell) has determined that Basic Auth over HTTP should not be supported Linux. the primary reasons include:

1: Both credentials and payload are sent un-encrypted.
2: Neither the client nor the server (omi) provide a method for disabling this combination.
3: It's too easy for users to use PSRP in an insecure manner.

As the above issue implies, this combination is currently broken in PowerShell on Linux. Instead of fixing it, the proposed change is to explicitly disable it and report a more appropriate error message.

The recommendation going forward will be to use Negotiate which results in the following:

From the OMI/Setup NTLM doc... 
- The password is hashed using an irreversable algorithm, so the password is more secure than Basic.
 - The client and server support encryption of your data over http connections, so SSL certificates are not required.

### PR DESCRIPTION
**Pull request by dantraMSFT**
## PR Summary

Disallow Basic Auth over HTTP on Unix

Fix: https://github.com/PowerShell/PowerShell/issues/6779

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/resources/RemotingErrorIdStrings.resx [hunk 1]
```diff
@@ -1110,6 +1110,9 @@ All WinRM sessions connected to PowerShell session configurations, such as Micro
   <data name="AuthenticationMechanismRequiresCredential" xml:space="preserve">
     <value>{0} authentication requires an explicit user name and password.  Specify the user name and password by using the -Credential parameter and try the command again.</value>
   </data>
+  <data name="BasicAuthOverHttpNotSupported">
+   <value>Basic authentication is not supported over HTTP on Unix</value>
```
Comment (markekraus) [inline src/System.Management.Automation/resources/RemotingErrorIdStrings.resx]: Missing `.` after `Unix`.

**Review by markekraus — CHANGES_REQUESTED** [review]

File: test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1 [hunk 2]
```diff
@@ -11,6 +11,26 @@ Describe "New-PSSession basic test" -Tag @("CI") {
...
+        $password = ConvertTo-SecureString -String "password" -AsPlainText -Force
+        $credential = [PSCredential]::new('username', $password)
+        try
+        {
+            New-PSSession -ComputerName 'localhost' -Credential $credential -Authentication Basic
+            throw "New-PSSession should throw"
+        }
+        catch
+         {
+            $_.Exception | Should -BeOfType [System.Management.Automation.Remoting.PSRemotingTransportException]
```
Comment (iSazonov) [inline test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1]: We should use 
```c#
$ex = { ... } | Should -Throw -PassThrou "FullyQualifiedErrorId"
$ex.Exception ...
```

Comment (TravisEz13) [reply test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1]: Example: https://github.com/PowerShell/PowerShell/blob/986a9eeecdb800cde928a0d9e7e926fe5086d4cf/test/powershell/Modules/Microsoft.PowerShell.Utility/Remove-Variable.Tests.ps1#L11

File: src/System.Management.Automation/resources/RemotingErrorIdStrings.resx (hunk 1 above)
Comment (dantraMSFT) [reply src/System.Management.Automation/resources/RemotingErrorIdStrings.resx]: Fixed

File: test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1 (hunk 2 above)
Comment (dantraMSFT) [reply test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1]: Fixed

**Review by iSazonov — APPROVED** [review]

**Review by markekraus — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**